pkg/controller/mutators/core/controller_core.go

```diff
@@ -0,0 +1,314 @@
+) *Reconciler {
+	r := &Reconciler{
+		system:     mutationSystem,
+		Client:     mgr.GetClient(),
+		tracker:    tracker,
+		getPod:     getPod,
+		scheme:     mgr.GetScheme(),
+		reporter:   ctrlmutators.NewStatsReporter(),
+		cache:      ctrlmutators.NewMutationCache(),
+		kind:       kind,
+		newObj:     newObj,
+		newMutator: newMutator,
+		log:        logf.Log.WithName("controller").WithValues(logging.Process, fmt.Sprintf("%s_controller", strings.ToLower(kind))),
+	}
+	if getPod == nil {
+		r.getPod = r.defaultGetPod
+	}
+	return r
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler.
+func add(mgr manager.Manager, r *Reconciler) error {
+	if !*mutation.MutationEnabled {
+		return nil
+	}
+
+	// Create a new controller
+	c, err := controller.New(fmt.Sprintf("%s-controller", strings.ToLower(r.kind)), mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the mutator
+	if err = c.Watch(
+		&source.Kind{Type: r.newObj()},
+		&handler.EnqueueRequestForObject{}); err != nil {
+		return err
+	}
+
+	// Watch for changes to MutatorPodStatus
+	err = c.Watch(
+		&source.Kind{Type: &statusv1beta1.MutatorPodStatus{}},
+		handler.EnqueueRequestsFromMapFunc(mutatorstatus.PodStatusToMutatorMapper(true, r.kind, util.EventPackerMapFunc())),
+	)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Reconciler reconciles mutator objects.
+type Reconciler struct {
+	client.Client
+	kind       string
+	newObj     func() client.Object
+	newMutator func(client.Object) (types.Mutator, error)
+
+	system   *mutation.System
+	tracker  *readiness.Tracker
+	getPod   func() (*corev1.Pod, error)
+	scheme   *runtime.Scheme
+	reporter ctrlmutators.StatsReporter
+	cache    *ctrlmutators.Cache
+	log      logr.Logger
+}
+
+// +kubebuilder:rbac:groups=mutations.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a mutator object and syncs it with the mutation system.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.log.Info("Reconcile", "request", request)
+	startTime := time.Now()
+
+	gvk := schema.GroupVersionKind{
+		Group:   mutationsv1alpha1.GroupVersion.Group,
+		Version: mutationsv1alpha1.GroupVersion.Version,
+		Kind:    r.kind,
+	}
+
+	deleted := false
+	mutatorObj := r.newObj()
+	err := r.Get(ctx, request.NamespacedName, mutatorObj)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+
+		deleted = true
+		mutatorObj = r.newObj()
+		mutatorObj.SetName(request.NamespacedName.Name)
+		mutatorObj.SetNamespace(request.NamespacedName.Namespace)
+		mutatorObj.GetObjectKind().SetGroupVersionKind(gvk)
+	}
+	deleted = deleted || !mutatorObj.GetDeletionTimestamp().IsZero()
+	tracker := r.tracker.For(gvk)
+
+	mID := types.MakeID(mutatorObj)
+	fmt.Println("MID IS", mID, "END")
```

Comment:
Is a print statement meant to be in here?

Reply:
Ooops, dangling debug code